<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Meta_Data_RUCCTS_BRC.knit</title>
    <meta charset="utf-8" />
    <meta name="author" content="Rockefeller University, BRC and RUCCTS Bioinformatics" />
    <script src="libs/header-attrs-2.17/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="metropolisCustom.css" type="text/css" />
    <link rel="stylesheet" href="metropolis-fontsCustom.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# MetaData in Bioinformtics Research
<html>
<div style="float:left">

</div>
<hr color='#EB811B' size=1px width=796px>
</html>
]
.author[
### Rockefeller University, BRC and RUCCTS Bioinformatics
]
.date[
### <a href="https://rockefelleruniversity.github.io/RU_reproducibleR_Abridged/" class="uri">https://rockefelleruniversity.github.io/RU_reproducibleR_Abridged/</a>
]

---





# &lt;font size="4"&gt;**Accessing information from a public database: GEO**&lt;/font&gt;
&lt;font size="4"&gt; Minimum Information About a Microarray Experiment (MIAME) is a set of guidelines developed by the Microarray Gene Expression Data (MGED). It been adopted by many journals to support an independent evaluation of results. Gene Expression Omnibus (GEO) database at the National Center for Biotechnology Information and European Bioinformatics Institute (EBI) ArrayExpress follow same standard. \

&lt;font size="4"&gt;  Among the several R packages that can be used to access its data, GEOquery is a simple and widely accepted package. \
To install



```r
if (!require("BiocManager", quietly = TRUE)) install.packages("BiocManager")
BiocManager::install("GEOquery", force = TRUE)
```







# &lt;font size="4"&gt;**Creating metadata using template: CEDAR Workbench**&lt;/font&gt;

&lt;font size="4"&gt; It is an online resource that makes it easy to collect and use metadata \

&lt;font size="4"&gt;**It helps user to**&lt;/font&gt; \
&lt;font size="4"&gt;create user-friendly, shareable forms for collecting metadata, with features like &lt;/font&gt; \

  * &lt;font size="4"&gt;share your forms and metadata &lt;/font&gt; \

  * &lt;font size="4"&gt;link your questions (fields) and possible answers (values) to controlled terms &lt;/font&gt; \

  * &lt;font size="4"&gt; view metadata responses meeting your search critieria, in several forms &lt;/font&gt; \

  * &lt;font size="4"&gt;use the Workbench Desktop interface to manage your content &lt;/font&gt;\

  * &lt;font size="4"&gt;enable intelligent metadata suggestions in your template by using a field’s Suggestions tab &lt;/font&gt; \


**Webpage**: [**cedar.metadatacenter.org**](https://cedar.metadatacenter.org) \

Click on the “Register” link to

 * Create an account \

 
![](imgs/cedar_signIn.png)&lt;!-- --&gt;


Once registered, 

 * Click on the Shared with Me link on the left side to find sample template \
 * Try creating your own \

![](imgs/cedar_webpage.png)&lt;!-- --&gt;

# &lt;font size="4"&gt;**cedarr: Accessing CEDAR using R**&lt;/font&gt; \

For installation:\

```r
devtools::install_github("earnaud/cedarr", dependencies = TRUE)
```

```
## checkmate (NA -&gt; 2.1.0) [CRAN]
## * checking for file ‘/tmp/Rtmp9xj2HF/remotes1d1c7669ee13/earnaud-cedarr-ddfd0d1/DESCRIPTION’ ... OK
## * preparing ‘cedarr’:
## * checking DESCRIPTION meta-information ... OK
## * checking for LF line-endings in source and make files and shell scripts
## * checking for empty or unneeded directories
## Omitted ‘LazyData’ from DESCRIPTION
## * building ‘cedarr_0.0.0.9000.tar.gz’
```
  
For help: \

```r
help("API response", package = "cedarr")
# ?cedarr::`API response`
```



    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
